### DISCUSSION TITLE
Change focus after creating new space

### DISCUSSION THREAD

**Discussion by amlanjlahkar** [asmvik/yabai#2076] [Q&A] [answered]
I'm trying to make yabai focus the new space after its creation.

In my config I've this:
```
yabai -m signal --add event=space_created \
    action="focus_space $YABAI_SPACE_ID"
```

focus_space script
```sh
#!/bin/sh

space_index=$(yabai -m query --spaces | jq ".[] | select(.id == $1) | .index")

yabai -m space --focus $space_index
```

I checked the error log and apparently `$YABAI_SPACE_ID` is being passed as null

**Comment by asmvik** [comment, marked as answer]
You need to escape the variable so that it is passed along as part of the script instead of being evaluated in the .yabairc script.

```
yabai -m signal --add event=space_created action="focus_space \$YABAI_SPACE_ID"
```

You may also need to include the full path to the focus_space script, unless it is located somewhere in your $PATH.
